File: 0704_Binary_Search/0704_Binary_Search.py
from typing import List

class Solution: 
    def search(self, nums: List[int], target: int) -> int:
        i = 0 # first index
        j = len(nums)-1 # last index

        while i <= j:
            mid = (i+j)//2

            # Found Target
            if nums[mid] == target:
                return mid

            # Go Left
            if nums[mid] > target:
                j = mid-1
            
            # Go Right
            else:
                i = mid+1

        # Could not find
        return -1

File: 0704_Binary_Search/0704_Binary_Search.swift
class Solution {
    func search(_ nums: [Int], _ target: Int) -> Int {
        var i = 0
        var j = nums.count

        while (i <= j) {
            var mid = Int((i+j)/2)

            // Found Target
            if nums[mid] == target {
                return mid
            }

            // Could Not Find
            if j-i == 1 {
                break
            }

            // Go Left
            if nums[mid] > target {
                j = mid
            }
            // Go Right
            else {
                i = mid
            }
        }

        return -1
    }
}